homeassistant/components/shelly/config_flow.py

```diff
@@ -142,7 +151,9 @@ async def async_step_zeroconf(self, zeroconf_info):
         self._abort_if_unique_id_configured({CONF_HOST: zeroconf_info["host"]})
         self.host = zeroconf_info["host"]
         # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
-        self.context["title_placeholders"] = {"name": zeroconf_info["host"]}
+        self.context["title_placeholders"] = {
+            "name": _remove_prefix(zeroconf_info["properties"]["id"])
```

Comment:
We shouldn't remove the prefix and just use straight up the ID

Reply:
I have several shelly switch 2.5 units which have the same mac prefix. Without removing "shelly", the part that's unique is hidden and it's impossible to differentiate them

Reply:
Note that I'm only removing it when it starts with "shellyswitch", and only removing the "shelly" part.

Reply:
but then how will one know it's the switch ?

Reply:
I'm only removing "shelly", so it still starts with "switch25"